src/components/shell/shell.js

```diff
@@ -66,9 +64,90 @@ class Shell extends Component {
             </Main>
           </div>
         }
-      </div>
+      </FluentShell>
     );
   }
+
+  getNavProps() {
+    const { pagesConfig, t, denyAccess } = this.props;
+    if (denyAccess || !pagesConfig) {
+      return null;
+    }
+
+    return {
+      isExpanded: this.state.isNavExpanded,
+      onClick: this.handleGlobalNavToggle,
+      attr: {
+        navButton: {
+          title: t(this.state.isNavExpanded ? 'globalNav.collapse' : 'globalNav.expand'),
+        },
+      },
+      children: pagesConfig.map((tabProps, i) => {
+        const label = t(tabProps.labelId);
+        return (
+          <NavLink key={i} to={tabProps.to} className="global-nav-item" activeClassName="global-nav-item-active" title={label}>
+            <Svg path={tabProps.svg} className="global-nav-item-icon" />
+            <div className="global-nav-item-text">{label}</div>
+          </NavLink>
+        );
+      })
+    };
+  }
+
+  getMastheadProps() {
+    const { pagesConfig, t, denyAccess, openSystemSettings, openUserProfile, openHelpFlyout, openFlyout } = this.props;
+    if (denyAccess) {
+      return {
+        branding: t('header.appName'),
+        more: {
+          title: t('header.more'),
+          selected: this.state.isMastheadMoreExpanded,
+          onClick: this.handleMastheadMoreToggle,
+        },
+      };
+    } else if (pagesConfig) {
+      return {
+        branding: t('header.appName'),
+        more: {
+          title: t('header.more'),
+          selected: this.state.isMastheadMoreExpanded,
+          onClick: this.handleMastheadMoreToggle,
+        },
+        toolbarItems: [{
+          icon: 'settings',
+          label: t('settingsFlyout.title'),
+          selected: openFlyout === 'settings',
+          onClick: openSystemSettings
+        },{
+          icon: 'help',
+          label: t('helpFlyout.title'),
+          selected: openFlyout === 'help',
+          onClick: openHelpFlyout
+        }, {
+          icon: 'contact',
+          label: t('profileFlyout.title'),
+          selected: openFlyout === 'profile',
+          onClick: openUserProfile
+        }]
+      };
+    } else {
+      return null; // no masthead
+    }
+  }
+
+  handleGlobalNavToggle = (e) => {
+    e && e.stopPropagation();
+    this.setState({
+        isNavExpanded: !this.state.isNavExpanded
+    });
+  }
+
+  handleMastheadMoreToggle = (e) => {
+    e.stopPropagation();
```

Comment:
Does `e` need to be checked here like on 139?

Reply:
We don't really need it at 139 either, but I'll add it in, just in case we need to call this manually for some a11y fixes.